software/ADPL_electron/ADPL_electron.ino

```diff
@@ -60,12 +59,7 @@ void loop() {
     temp_count = read_temp(temp_count);
 
     if ((currentTime - last_publish_time) > PUBLISH_DELAY) {
-        sprintf(temps_str,"HXCI:%.1f,HXCO:%.1f,HTR:%.1f,HXHI:%.1f,HXHO:%.1f,V:%d",
-                tempHXCI.temp, tempHXCO.temp, tempHTR.temp, tempHXHI.temp, tempHXHO.temp, int(valve.gasOn));
-        Particle.publish("TEMPS",temps_str);
-        delay(1000);
-        bucket.publish();
-        last_publish_time = currentTime;
+        last_publish_time = publish_data(last_publish_time);
```

Comment:
I also like how this will retry a "failed" publish on the next `loop()` iteration :D